### ISSUE TITLE
Error converting APNG to WEBP

### ISSUE THREAD

**Issue by sjbitcode** [python-pillow/Pillow#7015]
### What did you do?
Opened an animated png file and tried saving it as a webp image.

I tried two animated png's:
- without transparency ([source](https://lukeroberts.tv/blog/gif-vs-animated-png/))
![man](https://user-images.githubusercontent.com/6550256/225187639-bd483ce5-1dff-437b-a9c1-86c454e514d6.png)
- with transparency ([source](https://commons.wikimedia.org/wiki/File:Animated_PNG_example_bouncing_beach_ball.png))
![beach_ball](https://user-images.githubusercontent.com/6550256/225187697-c7932f7e-93bc-4569-b400-1e477781e0aa.png)

### What did you expect to happen?
I expected the webp versions to successfully save the animations for both images.

### What actually happened?
Error from `WebPImagePlugin.py` on save:
```pytb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sangeeta/Code/python-pillow-practice/venv/lib/python3.10/site-packages/PIL/Image.py", line 2431, in save
    save_handler(self, fp, filename)
  File "/Users/sangeeta/Code/python-pillow-practice/venv/lib/python3.10/site-packages/PIL/WebPImagePlugin.py", line 287, in _save_all
    enc.add(
TypeError: 'float' object cannot be interpreted as an integer
```

### What are your OS, Python and Pillow versions?

* OS: macOS Monterey v12.5
* Python: 3.10.2
* Pillow: 9.4.0

```python
from PIL import Image

# walking man apng
img = Image.open('img/man.png')
print(img.format)
print(img.mode)
print(img.get_format_mimetype())
print(img.info)
img.save('transformed/man.webp', save_all=True)


# beach ball apng
img2 = Image.open('img/beach_ball.png')
print(img2.format)
print(img2.mode)
print(img2.get_format_mimetype())
print(img2.info)
img2.save('transformed/beach_ball.webp', save_all=True)
```

The output:
```pycon
>>> img = Image.open('img/man.png')
>>> print(img.format)
PNG
>>> print(img.mode)
RGB
>>> print(img.get_format_mimetype())
image/apng
>>> print(img.info)
{'loop': 0, 'transparency': (0, 0, 16), 'bbox': (0, 0, 1400, 1050), 'duration': 41.666666666666664, 'disposal': 0, 'blend': 0}
>>> img.save('transformed/man.webp', save_all=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sangeeta/Code/python-pillow-practice/venv/lib/python3.10/site-packages/PIL/Image.py", line 2431, in save
    save_handler(self, fp, filename)
  File "/Users/sangeeta/Code/python-pillow-practice/venv/lib/python3.10/site-packages/PIL/WebPImagePlugin.py", line 287, in _save_all
    enc.add(
TypeError: 'float' object cannot be interpreted as an integer


>>> img2 = Image.open('img/beach_ball.png')
>>> print(img2.format)
PNG
>>> print(img2.mode)
RGBA
>>> print(img2.get_format_mimetype())
image/apng
>>> print(img2.info)
{'loop': 0, 'bbox': (0, 0, 100, 100), 'duration': 75.0, 'disposal': 1, 'blend': 0}
>>> img2.save('transformed/beach_ball.webp', save_all=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sangeeta/Code/python-pillow-practice/venv/lib/python3.10/site-packages/PIL/Image.py", line 2431, in save
    save_handler(self, fp, filename)
  File "/Users/sangeeta/Code/python-pillow-practice/venv/lib/python3.10/site-packages/PIL/WebPImagePlugin.py", line 287, in _save_all
    enc.add(
TypeError: 'float' object cannot be interpreted as an integer
```

**Comment by radarhere** [comment]
The error you are encountering can be resolved by either #6996 or #6977

**Comment by sjbitcode** [comment]
thanks for the quick reply @radarhere! Looking forward to seeing those changes merged in! 😄

**Comment by radarhere** [comment]
Oh, actually, if you want an immediate way to fix the error, you could just insert `img.info["duration"] = round(img.info["duration"])`.
```python
from PIL import Image

# walking man apng
img = Image.open('img/man.png')
print(img.format)
print(img.mode)
print(img.get_format_mimetype())
print(img.info)
img.info["duration"] = round(img.info["duration"])
img.save('transformed/man.webp', save_all=True)


# beach ball apng
img2 = Image.open('img/beach_ball.png')
print(img2.format)
print(img2.mode)
print(img2.get_format_mimetype())
print(img2.info)
img2.info["duration"] = round(img2.info["duration"])
img2.save('transformed/beach_ball.webp', save_all=True)
```

**Comment by sjbitcode** [comment]
Thanks @radarhere, that's a clever workaround!

This is probably a separate issue, but the `man.webp` image has a black background after the first frame; I've attached it here as a zip because I couldn't upload a webp image!
[man.webp.zip](https://github.com/python-pillow/Pillow/files/10984642/man.webp.zip)

I started reading through https://github.com/python-pillow/Pillow/issues/5755 and https://github.com/python-pillow/Pillow/issues/5837, maybe this is a color palette issue?

**Comment by radarhere** [comment]
I created PR #7018 to fix that part, applying transparency to the new frame before blending it with the current image.

**Comment by sjbitcode** [comment]
Just tested this, and it works perfectly! Thanks for the quick fix again!

**Comment by hugovk** [comment]
Yes, https://github.com/python-pillow/Pillow/pull/7018 has been merged and will be in 9.5.0 on 1st April 2023.

Thanks for the report.

**Comment by radarhere** [comment]
Pillow 9.5.0 has now been released.
